utils/hct/hctdb.py

```diff
@@ -3245,8 +3260,52 @@ def add_attr_arg(title_name, scope, args, doc):
         add_attr_arg("Unroll", "l", "Unroll the loop until it stops executing or a max count", [{"name":"Count", "type":"int"}])
         self.attributes = attributes
 
+class InstructionSignature:
```

Comment:
The general utility of this class seems pretty limited.  It only captures a processed subset of properties from a dxil operation.  Instead, you could write a wrapper class that captures the dxil operation upon construction and provides helper methods to retrieve/compute the properties that aren't as easy to retrieve directly.  Then you can iterate through the normal operation list, constructing the wrapper, filtering, and printing matches on the fly.  And filtering would still have access to all the properties of the original operation, instead of only what you deemed useful when writing this code.

I also don't think such a utility class belongs in hctdb.py.  If needed, it can go into hctdb_instrhelp.py.